### PR TITLE
feat(workflows/publish): golangci-lint

### PR DESCRIPTION
**Pull request by EtienneM**
`linter-*` and `tests` jobs are executed in parallel. In case of success, the `release` step is executed.

I tested to introduce a linter offense to ensure that it indeed fails in case of error, and the test was successful.

### CODE REVIEW

**Review by curzolapierre — CHANGES_REQUESTED** [review]
Overall it's LGTM

However I've small questions

Note for later, we could also separate the `linter-pull-request` and `linter-master` jobs by another trigger. Here, we use `on: [push]` for both, we may want to use `on: [pull-request]` in the future to simplify configuration.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -2,10 +2,63 @@ name: Test and Release Go CLI
...
+          check-latest: true
+      - name: Get golangci-lint configuration file
+        run: wget --output-document=$(pwd)/.golangci.yml https://sc-devtools.s3.eu-west-1.amazonaws.com/golang-ci/golangci.yml
+      - name: "Execute golangci-lint on the master branch"
+        uses: golangci/golangci-lint-action@v4
+        with:
+          # The `only-new-issues` flag is not working (https://github.com/golangci/golangci-lint-action/issues/531).
+          # We rather decided to use the suggestion from the FAQ (https://golangci-lint.run/usage/faq/#how-to-integrate-golangci-lint-into-large-project-with-thousands-of-issues) and use `--new` and `--new-from-rev`
+          # only-new-issues: false
+          args: "--config=$(pwd)/.golangci.yml --new --new-from-rev=HEAD~1"
```
[the remark this answers is not in the document]
Comment (EtienneM) [reply .github/workflows/publish.yml]: I actually misunderstood the use of `--new`. I removed the use of `--new` in f6aa960c. Here isthis option documentation:

> Show only new issues: if there are unstaged changes or untracked files, only those changes are analyzed

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -2,10 +2,63 @@ name: Test and Release Go CLI
...
+          check-latest: true
+      - name: Get golangci-lint configuration file
+        run: wget --output-document=$(pwd)/.golangci.yml https://sc-devtools.s3.eu-west-1.amazonaws.com/golang-ci/golangci.yml
+      - name: Get master branch commit ID
+        id: new-from-rev
+        run: echo "NEW_FROM_REV=$( git rev-parse origin/master )" >> "$GITHUB_OUTPUT"
+      - name: "Execute golangci-lint on a pull request"
+        uses: golangci/golangci-lint-action@v4
+        env:
+          NEW_FROM_REV: ${{ steps.new-from-rev.outputs.NEW_FROM_REV }}
```
[the remark this answers is not in the document]
Comment (EtienneM) [reply .github/workflows/publish.yml]: You are right it's useless. Removed in b57c7fb. Thanks :+1:

**Review by curzolapierre — APPROVED** [review]

**Comment by EtienneM** [comment]
With this PR being merged, I removed CodeShip from this project.

**Pull request merged**